auto_dev/data/templates/protocols/test_dialogues.jinja

```diff
@@ -0,0 +1,73 @@
+{{ header }}
+
+"""Test dialogues module for the {{ snake_name }} protocol."""
+
+from unittest.mock import MagicMock
+
+from pydantic import BaseModel, conint, confloat
+from hypothesis import given
+from hypothesis import strategies as st
+from aea.configurations.data_types import PublicId
+
+from packages.{{ author }}.protocols.{{ snake_name }}.dialogues import (
+    {{ camel_name }}Dialogue,
+    {{ camel_name }}Dialogues,
+)
+from packages.{{ author }}.protocols.{{ snake_name }}.message import {{ camel_name }}Message
+from packages.{{ author }}.protocols.{{ snake_name }}.tests.primitive_strategies import (
+    Int32,
+    Double,
+)
+from packages.{{ author }}.protocols.{{ snake_name }}.custom_types import (
+    {%- for custom_type in custom_types %}
+    {{ custom_type }},
+    {%- endfor %}
+)
+
+
+def shallow_dump(model: BaseModel) -> dict:
+    """Shallow dump pydantic model."""
+
+    return {name: getattr(model, name) for name in model.__fields__}
+
+
+def validate_dialogue(performative, model):
+    """Validate successful dialogue instantiation."""
+
+    skill_context = MagicMock()
+    skill_context.skill_id = PublicId(
+        name="mock_name",
+        author="mock_author",
+    )
+
+    dialogues = {{ camel_name }}Dialogues(
+        name="test_{{ snake_name }}_dialogues",
+        skill_context=skill_context,
+    )
+
+    dialogue = dialogues.create(
+        counterparty="dummy_counterparty",
+        performative=performative,
+        **shallow_dump(model),
+    )
+
+    assert dialogue is not None
```

Comment:
This allows us to abandon the [`BaseProtocolDialoguesTestCase`](https://github.com/valory-xyz/open-aea/blob/26738e20ca0937b5695e43bf2267525a475d8af4/aea/test_tools/test_protocol.py#L103-L168) and the scaffolded test code entirely. Yes, this code will thus be reproduced for every protocol (since not a base in any framework we import from) but it encapsulates all the logic of the base class and the scaffolded class by AEA-framework, with the added benefit of being 10x more interpretable.